### ISSUE TITLE
Code which runs with fastOptJS but crashes with fullOptJS

### ISSUE THREAD

**Issue by fdietze** [scala-js/scala-js#2786]
Minimal reproducing example project:
https://github.com/woost/wust2/tree/fullopt-bug

```bash
git clone https://github.com/woost/wust2 -b fullopt-bug --depth=1
cd wust2
./test.sh
```

**Comment by sjrd** [comment]
Confirmed.

The root cause is that Closure decides to rename our internal property `$classData` (with we use to distinguish Scala.js objects from JS objects) into `b`. And then of course it mistakes some JS objects for Scala.js objects, and things go awry.

We should prevent GCC from renaming `$classData`, although
1. It will increase the size of `-opt.js` files
2. It still leaves us vulnerable to a JS object that has a property named `$classData`.

A full solution would be to use a `symbol` instead of `$classData`. And when symbols are not available, a randomized name with sufficient entropy (e.g., that of a UUID).

**Comment by sjrd** [comment]
Obvious workaround (but which makes `-opt.js` files much bigger): disable GCC with
```scala
scalaJSOptimizerOptions in (Compile, fullOptJS) ~= { _.withUseClosureCompiler(false) }
```

**Comment by sjrd** [comment]
Fixed in 5918016b88f420c9678888450e2cbe6095f4581d.
